models/pull.go

```diff
@@ -352,6 +354,26 @@ func (pr *PullRequest) GetCommitMessages() string {
 	return stringBuilder.String()
 }
 
+// GetCommitDivergence get Divergence of a pull request
+func (pr *PullRequest) GetCommitDivergence() *git.DivergeObject {
+	return &git.DivergeObject{Ahead: pr.CommitsAhead, Behind: pr.CommitsBehind}
+}
+
+// UpdateCommitDivergence update Divergence of a pull request
+func (pr *PullRequest) UpdateCommitDivergence(div *git.DivergeObject) error {
```

Comment:
`models` depends on `git` is not a very good idea. I think we can just use `commitsAhead` and `commitsBehind` as two parameters of the method.

Reply:
fair point - change it

Reply:
* `func (pr *PullRequest) GetCommitMessages() string {}`
* `func (pr *PullRequest) GetLastCommitStatus() (status *CommitStatus, err error) {}`
* `func (pr *PullRequest) IsHeadEqualWithBranch(branchName string) (bool, error) {}`

has to move then too - I'll send a pull for it later ...

Reply:
@lunny -> https://github.com/go-gitea/gitea/pull/11025